src/exaerror.lua

```diff
@@ -0,0 +1,164 @@
+local M = {
+    VERSION = "1.0.0",
+}
+
+-- Lua 5.1 backward compatibility
+_G.unpack = table.unpack or _G.unpack
+
+---
+-- Convert error to a string representation.
+-- <p>
+-- Note that <code>__tostring</code> is the metamethod called by Lua's global <code>tostring</code> function.
+-- This allows using the error message in places where Lua expects a string.
+-- </p>
+--
+-- @return string representation of the error object
+--
+function M:__tostring()
+    local pieces = {}
+    if self.message then
+        table.insert(pieces, self:get_message())
+    end
+    if self.mitigations then
+        for _, mitigation in ipairs(self.mitigations) do
+            table.insert(pieces, mitigation)
+        end
+    end
+    if self.code then
+        if #pieces > 0 then
+            table.insert(pieces, 1, self.code .. ":")
```

Comment:
I'm not sure if you really need to handle this edge case. If someone forgets the message,  I think it's ok to have not so nice error message.
In java we do:
`E-TEST-1: Unkown Column !!!MISSING PARAMETER!!!`
That has the advantage that it catches your attention while testing

Maybe here you could do `E-TEST-1: No Error Message`

Reply:
I think the edge case is fine here. The message about a missing message leaves to much irritation on the end users part in my eyes.